File: csharp/DesignPattern.Interpreter/Context.cs
using System.Collections.Generic;
namespace DesignPattern.Interpreter
{
    /// <summary>
    /// 环境
    /// </summary>
    public class Context
    {
        private Dictionary<char, double> variable;
        public Dictionary<char, double> Variable 
        {
            get
            {
                if (this.variable == null)
                {
                    this.variable = new Dictionary<char, double>();
                }
                return this.variable;
            }
        }
    }
}

File: java/src/com/zhy/DesignPattern/Interpreter/Context.java
package com.zhy.DesignPattern.Interpreter;

import java.util.HashMap;

public class Context {

	private HashMap<Character, Double> variable;

	public HashMap<Character, Double> getVariable() {
		if(variable==null)
		{
			variable = new HashMap<Character, Double>();
		}
		return variable;
	}

	public void setVariable(HashMap<Character, Double> variable) {
		this.variable = variable;
	}
}
